### PR TITLE
Add example: Sequential OED with sigmoid-transformed linear model

### PR DESCRIPTION
**Pull request by ae-foster**

### CODE REVIEW

File: examples/contrib/oed/sequential_oed_sigmoid_lm.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import numpy as np
+
+import pyro
+from pyro import optim
+import pyro.distributions as dist
+from pyro.contrib.oed.eig import barber_agakov_ape
+from pyro.contrib.oed.util import rmv
+
+from models.bayes_linear import sigmoid_model, rf_group_assignments
+from guides.amort import SigmoidGuide
```
Comment (ae-foster) [inline examples/contrib/oed/sequential_oed_sigmoid_lm.py]: Change these two import lines, once #1371 merged

File: examples/contrib/oed/sequential_oed_sigmoid_lm.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            model = sigmoid_model(torch.tensor(0.),
+                                  torch.tensor([10., 2.5]),
+                                  torch.tensor(0.),
+                                  torch.tensor([1.]*5 + [10.]*5),
+                                  torch.tensor(1.),
+                                  100.*torch.ones(10),
+                                  1000.*torch.ones(10),
+                                  AB_sigmoid_design_6d)
+            my_guide = sigmoid_ba_guide(6)
+            ba_kwargs = {"num_samples": 100, "num_steps": 500, "guide": my_guide.guide,
```
Comment (ae-foster) [inline examples/contrib/oed/sequential_oed_sigmoid_lm.py]: remove `.guide`, this method was renamed to `forward`

File: examples/contrib/oed/sequential_oed_sigmoid_lm.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import pyro
+from pyro import optim
+import pyro.distributions as dist
+from pyro.contrib.oed.eig import barber_agakov_ape
+from pyro.contrib.oed.util import rmv
+
+from models.bayes_linear import sigmoid_model, rf_group_assignments
+from guides.amort import SigmoidGuide
+
+# Random effects designs
```
Comment (ae-foster) [inline examples/contrib/oed/sequential_oed_sigmoid_lm.py]: Add a couple of lines of docstrings, make martin happy

**Comment by ae-foster** [comment]
@eb8680 thanks for taking care of those changes from #1371. I've added a nice description so I think that should be everything taken care of
